src/backend/catalog/aclchk.c

```diff
@@ -4687,6 +4687,9 @@ recordExtensionInitPriv(Oid objoid, Oid classoid, int objsubid, Acl *new_acl)
 	if (!creating_extension && !binary_upgrade_record_init_privs)
 		return;
 
+	if (pltsql_check_store_init_privs_flag_hook && ((*pltsql_check_store_init_privs_flag_hook)()))
```

Comment:
I think we should directly check the GUC here instead of doing it in a hook. We can put the required checks in guc_assign_hook if needed.

Reply:
1. I was going to do the same earlier, but this will be cleaner to manage. Because there will be condition needed whether we are in creation/alter phase or not / then what is the value of guc / then skip if superuser.
2. No, we can't really put those checks only in guc_assign_hook because we also want to make sure that it is being utilised only while creation and alter.